src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -178,6 +183,105 @@ public static void Main()
         verifier.VerifyDiagnostics();
     }
 
+    [Fact]
+    public void FileLocalAttributeDefinitions_01()
+    {
+        var source = """
+            using System;
+            using System.Runtime.CompilerServices;
+
+            C.M();
+
+            class C
+            {
+                [Interceptable]
+                public static void M() => throw null!;
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 4, 3)]
+                public static void M()
+                {
+                    Console.Write(1);
+                }
+            }
+
+            namespace System.Runtime.CompilerServices
+            {
+                file class InterceptableAttribute : Attribute { }
+                file class InterceptsLocationAttribute : Attribute
+                {
+                    public InterceptsLocationAttribute(string filePath, int line, int character) { }
+                }
+            }
+            """;
+
+        var verifier = CompileAndVerify((source, "Program.cs"), expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void FileLocalAttributeDefinitions_02()
+    {
+        var source0 = """
+            using System.Runtime.CompilerServices;
+
+            public class C
+            {
+                [Interceptable]
+                public static void M() => throw null!;
+            }
+
+            namespace System.Runtime.CompilerServices
+            {
+                file class InterceptableAttribute : Attribute { }
+            }
+            """;
+
+        var source1 = """
+            using System;
+            using System.Runtime.CompilerServices;
+
+            C.M();
+
+            static class D
+            {
+                [InterceptsLocation("File1.cs", 4, 3)]
+                public static void M()
+                {
+                    Console.Write(1);
+                }
+            }
+
+            namespace System.Runtime.CompilerServices
+            {
+                file class InterceptsLocationAttribute : Attribute
+                {
+                    public InterceptsLocationAttribute(string filePath, int line, int character) { }
+                }
+            }
+            """;
+
+        var verifier = CompileAndVerify(new[] { (source0, "File0.cs"), (source1, "File1.cs") }, expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+
+        var comp0 = CreateCompilation((source0, "File0.cs"));
+        comp0.VerifyEmitDiagnostics();
+
+        var verifier1 = CompileAndVerify((source1, "File1.cs"), new[] { comp0.ToMetadataReference() }, expectedOutput: "1");
+        verifier1.VerifyDiagnostics();
+
+        // https://github.com/dotnet/roslyn/issues/67079
```

Comment:
nit: I'm assuming that issue will get fixed in `main` branch. If so, consider adding PROTOTYPE (not just the link) here.